File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int total = 0;
    
    public void helper(TreeNode current,boolean toLeft){
        if(current == null) return;
        
        if(current.left == null && current.right == null){
            if(toLeft){
               total+=current.val; 
            } 
            return;
        }
            
        helper(current.left,true);
        helper(current.right,false);
    }
    
    public int sumOfLeftLeaves(TreeNode root) {
        helper(root,false);
        return total;
    }
}

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumOfLeftLeaves = function(root) {
    let result = {sum: 0};
    
    if(root === null) return 0;
    else if(root.left === null && root.right === null) return 0;
    
    helper(root.left, true, result);
    helper(root.right, false, result);
    
    return result.sum;
};

function helper(current, isLeft, result) {
    if(current === null) return;
    else if(current.left === null && current.right === null) {
        if(isLeft) {
            result.sum += current.val; 
        }
        return;
    }
    
    helper(current.left, true, result);
    helper(current.right, false, result);
}